conans/model/info.py

```diff
@@ -575,6 +575,27 @@ def header_only(self):
         self.options.clear()
         self.requires.clear()
 
+    def msvc_compatible(self):
+        if self.settings.compiler != "msvc":
+            return
+
+        compatible = self.clone()
+        version = compatible.settings.compiler.version
+        runtime = compatible.settings.compiler.runtime
```

Comment:
```suggestion
        runtime = compatible.settings.get_safe("compiler.runtime")
```

Reply:
Nop, at this point, using the ``msvc`` compiler, the ``compiler.runtime`` must be defined, otherwise I want it to raise, not to continue silently. Why would you want/allow this to be undefined here? It is not possible in the settings (it doesn't have a ``None`` value)

Reply:
that's very possible `settings.yml` might be modified by the user, either manually, or by running `conan config install`. user may add None value or remove `compiler.runtime` sub-setting completely. IMO code should be reliable for custom settings, as we're allowing them and promote it as a first-class feature (especially for Enterprise users).

Reply:
So far this would be an unsupported case, the ``msvc`` compiler always has a ``runtime``, it is not possible to compile without it. What does a ``None`` value means? What is the use case? It is important to code new features initially more restrictive/limited scope, and learn from use cases, better than be relaxed, allow use cases that we completely ignore and that might have other implications. I strongly prefer to evolve from the basic "canonical" use case, and account for variants and special use cases later.